app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/MissionStartActivity.kt

```diff
@@ -0,0 +1,119 @@
+package ch.epfl.sdp.drone3d.ui.mission
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.ProgressBar
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import ch.epfl.sdp.drone3d.R
+import ch.epfl.sdp.drone3d.service.api.drone.DroneData.DroneStatus
+import ch.epfl.sdp.drone3d.service.api.drone.DroneData.DroneStatus.*
+import ch.epfl.sdp.drone3d.service.api.drone.DroneService
+import ch.epfl.sdp.drone3d.service.impl.drone.DroneUtils
+import ch.epfl.sdp.drone3d.ui.ToastHandler
+import ch.epfl.sdp.drone3d.ui.map.MissionInProgressActivity
+import com.mapbox.mapboxsdk.geometry.LatLng
+import dagger.hilt.android.AndroidEntryPoint
+import io.reactivex.disposables.Disposable
+import timber.log.Timber
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class MissionStartActivity : AppCompatActivity() {
+
+
+    companion object {
+        val MISSION_START_STATUS: List<DroneStatus> = listOf(
+            IDLE,
+            ARMING,
+            TAKING_OFF,
+            SENDING_ORDER,
+            STARTING_MISSION
+        )
+    }
+
+    @Inject lateinit var droneService : DroneService
+
+    private var disposable: Disposable? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_mission_start)
+
+        val progressBar = findViewById<ProgressBar>(R.id.mission_start_progress).apply {
+            max = totalProgress()
+        }
+
+        val progressText = findViewById<TextView>(R.id.mission_start_text)
+
+        droneService.getData().getDroneStatus().observe(this) { status ->
```

Comment:
Correct me if I'm wrong but, I belive that when you provide the activity to a observer, it stops observing when the class is destroyed automatically.

Reply:
I will onvestigate that

Reply:
That's true